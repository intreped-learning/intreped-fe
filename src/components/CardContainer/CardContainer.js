import React from 'react';
import { useSelector } from 'react-redux';
import CourseCard from '../CourseCard/CourseCard';
import './CardContainer.scss';

const CardContainer = () => {
  const { courses } = useSelector(state => state);
  const courseCards = courses.map(course => {
    return (
      <CourseCard
        key={course.id}
        id={course.id}
        category={course.category}
        title={course.title}
        description={course.description}
        thumbnail = {course.thumbnail}
      />
    )
  })

  return (
    <div className="card-container">
      {courseCards}
    </div>
  )
}

export default CardContainer;