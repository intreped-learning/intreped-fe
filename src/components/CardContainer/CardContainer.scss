.card-container {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr;
  grid-template-rows: auto;
  width: auto;
  margin: 45px;
  margin-top: 80px;
}