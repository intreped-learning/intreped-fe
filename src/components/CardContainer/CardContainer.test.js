import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk';
import { BrowserRouter as Router} from 'react-router-dom';
import CardContainer from './CardContainer';

const mockStore = configureMockStore([thunk]);

describe('CardContainer', () => {
  it('should match snapshot', () => {
    const store = mockStore({
      teacher: 'Mr. Feeny',
      courses: [],
      modalOpen: true
    });
    const getWrapper = () => mount(
      <Router>
        <Provider store={store}>
          <CardContainer/>
        </Provider>
      </Router>
    );
    const wrapper = getWrapper();
    expect(wrapper).toMatchSnapshot();
  });
});