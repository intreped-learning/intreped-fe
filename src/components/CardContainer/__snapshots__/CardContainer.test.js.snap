// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match snapshot 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <CardContainer>
        <div
          className="card-container"
        />
      </CardContainer>
    </Provider>
  </Router>
</BrowserRouter>
`;
